### PR TITLE
Rira 13.managing datasets

### PR DESCRIPTION
**Pull request by theNIGHTMAR3**
Done:
-Managing Datasets
-Define database connection
-Create dataset function
-Rework services classes
-Getters and setter for dataset
-Manage connection Neo4J
-Unit tests

### CODE REVIEW

**Comment by KrzysztofNazar01** [comment]
@mrwrob 
Should we solve the conflicts between the branches and then create a new PR?

**Comment by mrwrob** [comment]
@KrzysztofNazar01, no need for new PR. Just rebase, resolve conflicts and push commit. If it's in the same branch, it will appear in this PR.

File: graph_api/node/node_router.py [hunk 1]
```diff
@@ -24,11 +24,12 @@ class NodeRouter:
...
 
     @router.post("/nodes", tags=["nodes"], response_model=NodeOut)
-    async def create_node(self, node: NodeIn, response: Response):
+    async def create_node(self, node: NodeIn, database_name: str, response: Response):
         """
         Create node with optional labels
         """
-        create_response = self.node_service.save_node(node)
+        print("create_node Method!")
+        create_response = self.node_service.save_node(node, database_name)
```
Comment (mrwrob) [inline graph_api/node/node_router.py:32]: Is it some debugging artefact?

File: graph_api/database_service.py [hunk 2]
```diff
@@ -28,100 +28,149 @@ def __new__(cls):
...
             Result of request      
         """
+
+        # self.database_url = replace_db_name(self.database_url, database_name)
         commit_body = {
             "statements": [{"statement": statement}]
         }
-        response = self.post(commit_body)
+        response = self.post(commit_body, database_name)
+        print("response in post_statement: ", response)
```
Comment (mrwrob) [inline graph_api/database_service.py:48]: Is it needed here?

File: grisera_api/graph_api_service.py [hunk 3]
```diff
@@ -25,11 +30,21 @@ def post(self, url_part, request_body):
...
+        # print("###### URL_PART: ", url_part)
         response = requests.post(url=self.graph_api_url + url_part,
                                  json=request_body).json()
+        # print("###### RESPONSE: ", response)
+        # try:
+        #     response_data = response.json()
+        # except json.JSONDecodeError as json_error:
+        #     print(f"JSON decoding error occurred: {json_error}")
+        #     # Handle the JSONDecodeError
+        #     response_data = {}
```
Comment (mrwrob) [inline grisera_api/graph_api_service.py:43]: What's about these comments?

File: graph_api/dataset/dataset_router.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+        return create_response
+
+    @router.get("/datasets", tags=["datasets"], response_model=DatasetsOut)
+    async def get_datasets(self, response: Response):
+        """
+        Create directed and named dataset
+        """
+        #It don't have to be 'neo4j', any existing database will pass
+        database_name = "neo4j"
```
Comment (mrwrob) [inline graph_api/dataset/dataset_router.py:44]: Should the database name be passed through the call?

**Review by mrwrob — CHANGES_REQUESTED** [review]
I think we should add default value (`neo4j` I think) for  `database_name` in all routers in  `grisera_api`. Otherwise we'll break backward compatibility.  

If you want to log some issue on the console, you should not use prints: https://climbtheladder.com/8-fastapi-logging-best-practices/

**Pull request closed without merging**